qa/os/src/test/java/org/elasticsearch/packaging/util/Packages.java

```diff
@@ -240,6 +241,45 @@ private static void verifyDefaultInstallation(Installation es, Distribution dist
 
         Stream.of("users", "users_roles", "roles.yml", "role_mapping.yml", "log4j2.properties")
             .forEach(configFile -> assertThat(es.config(configFile), file(File, "root", "elasticsearch", p660)));
+        verifySecurityAutoConfigured(es, distribution);
+    }
+
+    private static void verifySecurityAutoConfigured(Installation es, Distribution distribution) throws IOException {
```

Comment:
I'd really rather not continue to add test coverage as assertions to `verifyInstallation()`. We should implement this as a new distinct test suite. We also need to add _a lot_ more coverage here for all the other scenarios as this only covers the "happy path". For example:

- Verify the correct behavior (still under debate) when `elasticsearch.yml` file is non-writable
- Verify we don't auto config security if security settings already exist
- Verify we don't auto config security on an upgrade

There are likely other more specific scenarios that probably are better implemented as unit tests for the CLI tool for which no tests currently yet exist.

Reply:
> We also need to add a lot more coverage here for all the other scenarios

Thanks for the feedback, I'll make sure I'll add test coverage before I raise this for review. 

> Verify the correct behavior (still under debate) when elasticsearch.yml file is non-writable

What is under debate regarding this ?

> There are likely other more specific scenarios that probably are better implemented as unit tests for the CLI tool for which no tests currently yet exist.

Makes sense, I think it's better if we keep comments targeted in relevant PRs so that we don't miss anything.

Reply:
> What is under debate regarding this ?

Like I mentioned, @colings86 and I briefly talked about possibly erroring on startup if we fail to auto-config security rather than pressing forward. No decisions were made but we might want to reconfirm that this is indeed the behavior we want.